edgelet/iotedge/src/check/checks/identity_certificate_expiry.rs

```diff
@@ -0,0 +1,156 @@
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+
+use failure::{self, Context, ResultExt};
+
+use edgelet_core::{self, AttestationMethod, ManualAuthMethod, Provisioning, RuntimeSettings};
+
+use crate::check::{checker::Checker, Check, CheckResult};
+
+#[derive(Default, serde_derive::Serialize)]
+pub(crate) struct IdentityCertificateExpiry {
+    provisioning_mode: Option<&'static str>,
+    certificate_info: Option<CertificateValidity>,
+}
+
+impl Checker for IdentityCertificateExpiry {
+    fn id(&self) -> &'static str {
+        "identity-certificate-expiry"
+    }
+    fn description(&self) -> &'static str {
+        "production readiness: identity certificates expiry"
+    }
+    fn execute(&mut self, check: &mut Check) -> CheckResult {
+        self.inner_execute(check)
+            .unwrap_or_else(CheckResult::Failed)
+    }
+    fn get_json(&self) -> serde_json::Value {
+        serde_json::to_value(self).unwrap()
+    }
+}
+
+impl IdentityCertificateExpiry {
+    fn inner_execute(&mut self, check: &mut Check) -> Result<CheckResult, failure::Error> {
+        let settings = if let Some(settings) = &check.settings {
+            settings
+        } else {
+            return Ok(CheckResult::Skipped);
+        };
+
+        match settings.provisioning() {
+            Provisioning::Dps(dps) => {
+                self.provisioning_mode = Some("Dps");
```

Comment:
`"dps"` (lower-case).

Actually, we do want to know what kind of DPS provisioning it is too, so set it to `dps-x509` inside the block at line 44 and `dps-other` otherwise.